src/main.cpp

```diff
@@ -136,7 +119,24 @@ static void get_options(int argc, char **argv, po::variables_map &options) {
   desc.add(expert);
 
   po::store(po::parse_command_line(argc, argv, desc), options);
-  po::store(po::parse_environment(desc, environment_option_name), options);
+  po::store(po::parse_environment(desc,
+    [desc](const std::string &name) {
+          std::string option;
+          // convert an environment variable name to an option name
+          if (name.substr(0, 7) == "CGIMAP_") {
+            std::transform(name.begin() + 7, name.end(),
+                           std::back_inserter(option),
+                           [](unsigned char c) {
+                           return c == '_' ? '-' : std::tolower(c);
+                   });
+            for (const auto &d : desc.options()) {
+              if (d->long_name() == option)
+                return option;
+            }
+            std::cout << "Ignoring unknown environment variable: " << name << std::endl;
```

Comment:
Validating against all option names from the `options_description` object `desc`.